### PR TITLE
Update to v0.24.0 and enable gym-mujoco

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/conda-forge/gym-feedstock/pull/25 .
Fix https://github.com/conda-forge/gym-feedstock/issues/7 .

I removed `gym-robotics` as since 0.22 it was moved to a separate package (see https://github.com/openai/gym/releases/tag/0.22.0, https://github.com/Farama-Foundation/Gym-Robotics).

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/gym the PR is ready for review.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -111,39 +108,18 @@ outputs:
...
         - python
         - {{ pin_subpackage('gym', exact=True) }}
-        - mujoco-py >=1.50,<2.0
+        - mujoco-python
         - imageio
     test:
       imports:
         - gym.envs.mujoco
 
-  - name: gym-robotics
```
Comment (h-vetinari) [inline recipe/meta.yaml:130]: Has this (sub-)package been removed upstream?

**Review by h-vetinari — COMMENTED** [review]
Looks good, just one question.

Thanks a lot for tackling this! :)

File: recipe/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipe/meta.yaml:130]: Yes, there are a few links in the PR description on this.

Comment (h-vetinari) [reply recipe/meta.yaml:130]: Ok, sounds good

**Comment by traversaro** [comment]
Thanks @h-vetinari !

**Comment by h-vetinari** [comment]
I had assumed you had checked upstream for dependency changes... now that #27 rolls around and I check, it seems we're pretty far off base with the old deps, to the point where I think we'll have to mark 0.24.0 as broken. my bad for assuming, but sad face.

**Comment by h-vetinari** [comment]
In fact, it makes extensive [use of](https://github.com/openai/gym/blob/0.24.1/setup.py#L18-L22) `pygame`, which hasn't been packaged for conda-forge (though there have been two failed attempts already: https://github.com/conda-forge/staged-recipes/pull/13759 & https://github.com/conda-forge/staged-recipes/pull/16612)

**Comment by h-vetinari** [comment]
Also looks like mujoco needs some version pins (or a comment why we're not following upstream).

**Comment by traversaro** [comment]
> I had assumed you had checked upstream for dependency changes... now that #27 rolls around and I check, it seems we're pretty far off base with the old deps, to the point where I think we'll have to mark 0.24.0 as broken. my bad for assuming, but sad face.

Yes, sorry about that, I also assumed that the `import <..>` checks were enough to validate that the installed dependency were correct, but clearly I was wrong, sorry for the trouble.

**Comment by h-vetinari** [comment]
> Yes, sorry about that, I also assumed that the `import <..>` checks were enough to validate that the installed dependency were correct, but clearly I was wrong, sorry for the trouble.

It happens 🤷 
That's why it's good to be able to mark packages as broken. 🙃

**Pull request merged**